app/src/main/java/org/fossasia/pslab/activity/SensorActivity.java

```diff
@@ -3,25 +3,321 @@
 import android.os.Bundle;
 import android.support.annotation.Nullable;
 import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+import android.widget.Spinner;
+import android.widget.TextView;
 
 import org.fossasia.pslab.R;
 import org.fossasia.pslab.communication.ScienceLab;
+import org.fossasia.pslab.communication.peripherals.I2C;
+import org.fossasia.pslab.communication.peripherals.SPI;
+import org.fossasia.pslab.communication.sensors.ADS1115;
+import org.fossasia.pslab.communication.sensors.BH1750;
+import org.fossasia.pslab.communication.sensors.BMP180;
+import org.fossasia.pslab.communication.sensors.HMC5883L;
+import org.fossasia.pslab.communication.sensors.MLX90614;
+import org.fossasia.pslab.communication.sensors.MPU6050;
+import org.fossasia.pslab.communication.sensors.MPU925x;
+import org.fossasia.pslab.communication.sensors.SHT21;
+import org.fossasia.pslab.communication.sensors.Sx1276;
+import org.fossasia.pslab.communication.sensors.TSL2561;
 import org.fossasia.pslab.others.ScienceLabCommon;
 
-import butterknife.ButterKnife;
+import java.io.IOException;
+import java.util.ArrayList;
+
 
 /**
  * Created by asitava on 18/6/17.
  */
 
 public class SensorActivity extends AppCompatActivity {
+
+    private I2C i2c;
+    private SPI spi;
     private ScienceLab scienceLab;
 
+
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_sensor);
-        scienceLab = ScienceLabCommon.getInstance().scienceLab;
-        ButterKnife.bind(this);
+
+        scienceLab = ScienceLabCommon.scienceLab;
+        i2c = scienceLab.i2c;
+        spi = scienceLab.spi;
+
+        Button buttonSensorAutoscan = (Button) findViewById(R.id.button_sensor_autoscan);
+        Button buttonSensorGetRaw = (Button) findViewById(R.id.button_sensor_getraw);
+        final TextView tvSensorScan = (TextView) findViewById(R.id.tv_sensor_scan);
+        final TextView tvSensorGetRaw = (TextView) findViewById(R.id.tv_sensor_getraw);
+        final Spinner spinnerSensorSelect = (Spinner) findViewById(R.id.spinner_sensor_select);
+
+        buttonSensorAutoscan.setEnabled(false); // temporary, will be removed after testing
+        buttonSensorGetRaw.setEnabled(false);
+
+        buttonSensorAutoscan.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                ArrayList<Integer> data = new ArrayList<Integer>();
+                ArrayList<String> datastr = new ArrayList<String>();
+                String datadisp = "";
+                try {
+                    i2c.init();
```

Comment:
I don't have any I2C sensors as of now to actually try out the code but I did order `MPU6050` and `HMC5883L` . I will test this soon.